libvast/include/vast/store.hpp

```diff
@@ -36,20 +33,36 @@ class passive_store {
   /// @returns The number of rows in all contained slices.
   [[nodiscard]] virtual uint64_t num_events() const = 0;
 
-  /// Handle a query; this has a default implementation.
-  /// @param query_context The query context.
-  /// @param self The executing actor's self pointer.
-  /// @returns The number of rows in all contained slices.
-  [[nodiscard]] virtual caf::expected<uint64_t>
-  lookup(system::store_actor::pointer self,
-         const query_context& query_context) const;
+  /// Retrieve the schema associated with the data in the store.
+  /// @returns The VAST schema of the stored data.
+  [[nodiscard]] virtual type schema() const;
+
+  /// Execute a count query against the store.
+  /// @param expr The expression to filter events.
+  /// @param selection Pre-filtered ids to consider.
+  /// @return
```

Comment:
The @returns here is kinda important to fill out, as we expect users to implement this.

We should also mention that this has a default implementation.